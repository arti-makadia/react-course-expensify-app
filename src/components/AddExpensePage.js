import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpence } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpence(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpence: (expense) => dispatch(startAddExpence(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
