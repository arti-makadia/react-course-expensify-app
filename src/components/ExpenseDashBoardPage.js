import React from 'react';
import ExpenseList from './ExpenseList.js';
import ExpenseListFilters from './ExpenseListFilters.js';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashBoardPage = () => {
    return(
        <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
    );
}

export default ExpenseDashBoardPage;