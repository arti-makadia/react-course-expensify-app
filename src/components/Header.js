import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth.js';

export const Header = ({ startLogout }) => {
    return(
        <div>            
            <h1>
                Expensify
            </h1>
           
            <NavLink to='/dashboard' activeClassName="is-active">Home</NavLink>
            <NavLink to='/add' activeClassName="is-active">Create</NavLink>
            <NavLink to='/help' activeClassName="is-active">Help</NavLink>
            <button onClick={startLogout}>Logout</button>
        </div>        
   );
    };

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);