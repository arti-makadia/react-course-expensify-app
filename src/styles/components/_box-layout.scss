.box-layout {
    background: url('/images/bg.jpg');
    background-size: cover;
    height: 100vh;   /* View port height  */
    width: 100vw;    /* View port width  */
    align-items: center;  //set login box in centre
    display: flex;
    justify-content: center;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);  //transparent box amount set 0-1
    border-radius: 3px;
    padding: $l-size $m-size;   //l-size for top & bottom, m-size for left & right
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}