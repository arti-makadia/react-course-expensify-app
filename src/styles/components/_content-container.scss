.content-container {    
    margin: 0 auto;  // top-bottom=0 left-right=auto set margin according to browser size
    padding: 0 $m-size; //top-0, others are m-size
    max-width: 80rem;
}