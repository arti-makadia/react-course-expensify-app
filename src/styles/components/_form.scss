.form {
    display: flex;
    flex-direction: column; 
    >* {   // apply to all children tag in form tag
        margin-bottom: $m-size;
    }   
}

.form__error {
    margin: 0 0 $m-size;
    font-style: italic;
}