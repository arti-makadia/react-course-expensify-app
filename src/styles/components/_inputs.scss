.text-input {
    border: 1xp solid #cacccd;
    height: 50px;  //set universal class (*) into base.scss file for text-inputs
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}

.select {
    @extend .text-input;   //inherite all style of .text-input
   // padding: 1.5rem;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}