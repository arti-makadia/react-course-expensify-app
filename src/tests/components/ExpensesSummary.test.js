import React from 'react';
import { shallow } from 'enzyme';
import { expensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<expensesSummary expenseCount={1} expensesTotal={3478} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<expensesSummary expenseCount={23} expensesTotal={6345645744879} />);
    expect(wrapper).toMatchSnapshot();
});