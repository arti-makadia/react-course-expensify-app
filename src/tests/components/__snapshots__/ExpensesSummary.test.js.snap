// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<expensesSummary
  expenseCount={1}
  expensesTotal={3478}
/>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<expensesSummary
  expenseCount={23}
  expensesTotal={6345645744879}
/>
`;
